### DISCUSSION TITLE
How to contribute / donate / support Rechart devs?

### DISCUSSION THREAD

**Discussion by dsldsl** [recharts/recharts#2970] [General]
Hi.  Thank you to everyone who's worked on this project over the years.

I'm wondering if there are a group of "core" maintainer or people that I could contribute / support / donate to for their work on this project.  Doesn't look like it's too active these days, but would like to support if possible.  (i.e. in Github there is functionality to turn on a donate section, would love to for @xile611 @arcthur etc.)

Separately, I'd be interested in hiring someone who's active and well understands this lib for some small consulting hours dealing with nuanced questions.  🙏

**Comment by arcthur** [comment]
hi dsldsl

thanks for your advise. actually, i have lots of work and no time to support some issues and develop. if someone was interesting in this repo and have much time and ability to contribute new feature or handle issues, i will talk with him/her to develop

**Reply by nikolasrieble** [reply]
@dillonreedy is interested, and I am as well. @arcthur

**Reply by dillonreedy** [reply]
Yes please

**Reply by ckifer** [reply]
Also @dangreen in this thread https://github.com/recharts/recharts/issues/1833#issuecomment-1084362325

@arcthur - as someone who's been off and on active in the recharts triage team for the past few years and loves the project, I am willing to take ownership and delegate from here. I haven't contributed much code, but I understand it and am willing to orchestrate a bit to create progress.

**Reply by ckifer** [reply]
@arcthur tagging to ask again for a conversation surrounding making someone more active a Maintainer. Thanks!

**Comment by dsldsl** [comment]
Thanks @arcthur.  Are any "core" developers still active on this project?  Partially?  Def good to start looking for others to take on more pro active roles.

**Comment by dsldsl** [comment]
@arcthur do you know if it's possible to get ahold of @xile611 to transfer ownership of the repo to new people?  Or can you help with that?

**Reply by ckifer** [reply]
I've emailed him multiple times over the years with no responses :/

**Reply by igorlukanin** [reply]
I've been emailing as well, same outcome.

**Comment by ckifer** [comment]
Tagging all those under the `recharts` github org (other than those already in the discussion). Thumbs up if you would like to continue contributions if this repo becomes active again, thumbs down if not.

@avindra @evilucifero @huxiaoyun @jasonHzq @jasonslyvia @sainthkh @semoal @twobin

**Comment by dsldsl** [comment]
@arcthur @jasonHzq @evilucifero can we get your presence again to discuss transfer of ownership of this project?

Alternatively, the next plan would be to fork and continue dev in a new repo and with a new website.  Not ideal but better than death.

**Comment by dsldsl** [comment]
Could also start implementing updates in a new repo and then in the future hopefully offer them as pull requests here when we've made contact

**Reply by ckifer** [reply]
This is a really good idea, once we get 1 person with admin rights then we can go to town. It'd be a shame to lose the stars, popularity, visibility, etc by being in a fork. Down to start working on this a bit.

**Reply by ckifer** [reply]
Maybe someone with a stake in the game would like to create a new org or add a fork to their existing org (recharts-restored? restart-recharts? lol idk). Once that is done and maintainers are added we can move forward a bit. I'd create one but I don't really want to own naming, config, etc. without consensus.

**Reply by nikolasrieble** [reply]
@dillonreedy and I talked about forking as well. But we decided to first start with collecting an updated and improved version of recharts documentation in storybook.  https://github.com/dillonreedy/recharts-storybook

Right now the documentation is spread over the website / in the demo / in stackoverflow / in github issues.

**Reply by nikolasrieble** [reply]
As we progress on this repository, continuously investing and showing our dedication to the project, we will reach out to the core devs again to ask for contributor rights.
If that does not work, we will consider forking. 

Thus, for now, I suggest that we all get together to improve on recharts examples in the storybook repo created by @dillonreedy

**Reply by nikolasrieble** [reply]
Would you be interested to join forces? @dsldsl @dangreen

**Reply by dangreen** [reply]
Yes

**Reply by igorlukanin** [reply]
@ckifer @nikolasrieble @dsldsl I would definitely love to be a part of this, together with @dangreen. We're maintaining, supporting with code and docs contributions, and sometimes financially a number of prominent charting libraries and their integrations with front-end frameowrks (Chart.js, Chartist, canvg, etc.). So we'd like to be a part of Recharts revival and help however we can! 💌

**Reply by nikolasrieble** [reply]
Wonderful! Lets get together on the storybook and create a concise and complete documentation. I believe that some advanced examples of using recharts (i.e. including wrapper components) could lead the way to new feature development and improved default behaviour. 

I am currently working on a blog series in which I will share how we use recharts at Tulip Interfaces.

**Reply by dsldsl** [reply]
I don't understand what you're proposing - take a team of people who are actively interested and write docs? I def see value in that and appreciate all who contribute in any way.  And, I think active development with the codebase is critical Vs just documentation.  The maintainers seem to have moved on and without developer engagement this project will struggle to stay relevant.

**Reply by nikolasrieble** [reply]
@dsldsl As you can see in the commit history, recent changes have been merged 29 days ago. https://github.com/recharts/recharts/commits?author=arcthur

Further, non core developers had their PRs merged, such as https://github.com/recharts/recharts/pull/2900

Thus I do not conclude that the core devs moved on.

Instead I suggest to show investment and motivation to become an active maintainer of the project by solving a single problem - outdated documentation - in a separate repository. 

Yet I am hopeful that this might convince some of the core devs to add more contributors. 
If we fail to convince them, I am confident we should fork.

**Reply by nikolasrieble** [reply]
If you prefer to start forking right away, cool! Happy to contribute there as well. For now I will focus on storybook.

**Comment by ckifer** [comment]
Bump @arcthur

**Reply by ckifer** [reply]
Bump @arcthur @xile611

**Comment by Yilun-Sun** [comment]
I am willing to contribute @arcthur

**Comment by arcthur** [comment]
I haven't been on github for a while :>. I create a slack, if anyone want to contribute the repo, maybe join this https://join.slack.com/t/recharts/shared_invite/zt-1l5lal1py-4tfkgCqrdzCaE7t7_yUGWw

**Reply by ckifer** [reply]
Thanks for the response!! Joined

**Reply by ckifer** [reply]
Just tagging those that would want to join. @dsldsl @dangreen @dillonreedy @igorlukanin
